### DISCUSSION TITLE
Save `pyvista.Spline` to Abaqus `.inp` file with `pyvista.save_meshio`

### DISCUSSION THREAD

**Discussion by adamgranthendry** [pyvista/pyvista#1435] [Q&A] [answered]
I am trying to save a `pyvista.Spline` spline as an Abaqus input deck (`*.inp` file). When I try to use `pyvista.save_meshio` it gives me an error

```
meshio does not support VTK type 4
```

Looking at the `meshio` source code, type 4 is a `poly_line`. How can I export my spline to a format that can be read by Abaqus?

**Comment by tkoyama010** [comment, marked as answer]
@adamgranthendry If you want to export mesh file of FEM you have to make 1 cell with 2 points (or 1 cell with 3 points for second order cell).  The best way is converting your `pyvista.Spline` object to  `pyvista.Line`  objects.

```python
import pyvista as pv
import numpy as np


def make_points():
    """Helper to make XYZ points"""
    theta = np.linspace(-4 * np.pi, 4 * np.pi, 100)
    z = np.linspace(-2, 2, 100)
    r = z ** 2 + 1
    x = r * np.sin(theta)
    y = r * np.cos(theta)
    return np.column_stack((x, y, z))


points = make_points()


def lines_from_points(points):
    """Given an array of points, make a line set"""
    poly = pv.PolyData()
    poly.points = points
    cells = np.full((len(points) - 1, 3), 2, dtype=np.int_)
    cells[:, 1] = np.arange(0, len(points) - 1, dtype=np.int_)
    cells[:, 2] = np.arange(1, len(points), dtype=np.int_)
    poly.lines = cells
    return poly


line = lines_from_points(points)


pv.save_meshio("spline.inp", line, file_format="abaqus")
```

I made a script from ["Creating a Spline" example](https://docs.pyvista.org/examples/00-load/create-spline.html)
